datastream/dataset.py

```diff
@@ -89,8 +89,15 @@ def from_dataframe(dataframe: pd.DataFrame) -> Dataset[pd.Series]:
             get_item=lambda df, index: df.iloc[index],
         )
 
-    def __getitem__(self: Dataset[T], index: int) -> T:
-        '''Get an example ``T`` from the ``Dataset[T]``'''
+    def __getitem__(self: Dataset[T], index: Union[int, slice]) -> Union[T, Dataset[T]]:
+        '''Get an example or slice from the ``Dataset[T]``'''
+        if isinstance(index, slice):
```

Comment:
Doesn't it make more sense to check for an `int` and then let the slice be any iterable type instead? Maybe slice doesn't count as an iterable though but maybe we should add that then?